lib/pharos/config.rb

```diff
@@ -66,7 +66,7 @@ def dns_replicas
 
     # @return [Array<Pharos::Configuration::Node>]
     def master_hosts
-      @master_hosts ||= hosts.select { |h| h.role == 'master' }
+      @master_hosts ||= hosts.select { |h| h.role == 'master' }.sort_by(&:master_sort_score)
```

Comment:
This will only sort once and won't change even if sort scores change. I think it wouldn't matter if it sorted every time without memoization (or the `checks` could invalidate the cache on change, but that's a bit premature optimization)

Reply:
I made almost the same changes in #855 but kept the memoizing `master_hosts` and added `sorted_master_hosts` with sorting that does not memoize.

Reply:
I don't see why the sort order would change within `pharos up` run? AFAIK it's safe to memoize

Reply:
There's no score until gather_facts is run

Reply:
Reason for this refactoring is that we picked the wrong master in some cases (sorting was done in ClusterManager). At least now the logic is in one place and can be easily tweaked (if needed).

Reply:
This will never select the healthiest master since the sorted list is memoized before checks are made, so the sorting has no effect.

Reply:
That's a bit unfortunate (checked and yes it is memoized before gather facts) 😄

Reply:
Oh, I though the checks would've made earlier. In that case I revert my statement, we shouldn't memoize :D

Reply:
Fixed.